### PR TITLE
improve a11y for alert/banner components

### PR DESCRIPTION
**Pull request by radnan**

### CODE REVIEW

File: packages/react/src/banner/Banner.tsx [hunk 1]
```diff
@@ -31,28 +33,38 @@ export const Banner = forwardRef<HTMLDivElement, BannerProps>(
...
+
     return (
-      <Flex ref={ref} {...styles.banner({ colorScheme }, className)} {...props}>
-        <Icon asChild {...styles.icon({ colorScheme })}>
-          {createElement(iconMap[colorScheme])}
-        </Icon>
+      <BannerContextProvider descriptionId={descriptionId} labelId={labelId}>
+        <Flex
+          ref={ref}
+          role="alert"
```
Comment (toha993) [inline packages/react/src/banner/Banner.tsx:43]: Is alert the correct role for it vai? or we can use `role="banner"` here?

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role

Comment (radnan) [reply packages/react/src/banner/Banner.tsx:43]: no - banner is basically a replacement for the `<header>` element - and is more suited for a global header/topbar or `Skip to content` link

Comment (toha993) [reply packages/react/src/banner/Banner.tsx:43]: Got it vai.

File: packages/react/src/banner/Banner.tsx [hunk 2]
```diff
@@ -31,28 +33,38 @@ export const Banner = forwardRef<HTMLDivElement, BannerProps>(
...
+    const labelId = useId();
+
     return (
-      <Flex ref={ref} {...styles.banner({ colorScheme }, className)} {...props}>
-        <Icon asChild {...styles.icon({ colorScheme })}>
-          {createElement(iconMap[colorScheme])}
-        </Icon>
+      <BannerContextProvider descriptionId={descriptionId} labelId={labelId}>
+        <Flex
+          ref={ref}
```
Comment (toha993) [inline packages/react/src/banner/Banner.tsx:42]: Don't we need this for banner also vai? 

```
 aria-describedby={descriptionId}
 aria-labelledby={labelId}
```

Comment (radnan) [reply packages/react/src/banner/Banner.tsx:44]: oops - added

**Pull request merged**